site/pages/api/duplicate-disruption.api.ts

```diff
@@ -78,24 +92,24 @@ const duplicateDisruption = async (req: NextApiRequest, res: NextApiResponse): P
             },
             session.orgId,
             session.isOrgStaff,
-            template === "true",
         );
 
         if (draftDisruption.consequences) {
             await Promise.all(
                 draftDisruption.consequences.map(async (consequence) => {
-                    await upsertConsequence(consequence, session.orgId, session.isOrgStaff, template === "true");
+                    await upsertConsequence(consequence, session.orgId, session.isOrgStaff);
                 }),
             );
         }
 
-        redirectToWithQueryParams(
-            req,
-            res,
-            template === "true" ? ["template"] : [],
-            `${REVIEW_DISRUPTION_PAGE_PATH}/${newDisruptionId}`,
-            ["duplicate=true"],
+        const returnPath = encodeURIComponent(
+            `${DISRUPTION_DETAIL_PAGE_PATH}/${
+                templateId as string
+            }?template=true&return=${VIEW_ALL_TEMPLATES_PAGE_PATH}`,
         );
+        createDisruptionFromTemplate
+            ? redirectTo(res, `${CREATE_DISRUPTION_PAGE_PATH}/${newDisruptionId}?return=${returnPath}`)
```

Comment:
you could use the redirect with query params function here

Reply:
done